source/common/event/BUILD

```diff
@@ -34,24 +34,22 @@ envoy_cc_library(
         ":dispatcher_includes",
         ":libevent_scheduler_lib",
         ":real_time_system_lib",
-        ":signal_lib",
         ":scaled_range_timer_manager_lib",
+        ":signal_lib",
         "//envoy/common:scope_tracker_interface",
         "//envoy/common:time_interface",
         "//envoy/event:signal_interface",
         "//envoy/network:listen_socket_interface",
         "//envoy/network:listener_interface",
         "//source/common/common:assert_lib",
         "//source/common/common:thread_lib",
+        "//source/common/config:utility_lib",
         "//source/common/filesystem:watcher_lib",
-        "//source/common/network:dns_lib",
         "//source/common/network:connection_lib",
         "//source/common/network:listener_lib",
+        "//source/extensions/network/dns_resolver/cares:config",
```

Comment:
As discussed,  cares:config is depending on DNS factory. if removing cares:config from here, and add it as the dependancy of dns_factory_interface, it will create circular dependency.